gnd/src/main/java/com/google/android/gnd/persistence/remote/firestore/schema/LayerConverter.java

```diff
@@ -20,25 +20,28 @@
 
 import android.util.Log;
 import com.google.android.gnd.model.layer.Layer;
-import com.google.android.gnd.model.layer.Style;
+import java8.util.Optional;
 
 /** Converts between Firestore documents and {@link Layer} instances. */
 class LayerConverter {
   private static final String TAG = LayerConverter.class.getSimpleName();
 
   /** Black marker used when default remote data is corrupt or missing. */
-  private static final Style DEFAULT_DEFAULT_STYLE = Style.builder().setColor("#000000").build();
+  private static final String FALLBACK_COLOR = "#000000";
 
   static Layer toLayer(String id, LayerNestedObject obj) {
     Layer.Builder layer = Layer.newBuilder();
+    // Use "color" field until web UI is updated:
+    // TODO(https://github.com/google/ground-platform/issues/402): Remove fallback once updated in
+    // web client.
+    StyleNestedObject style =
+        Optional.ofNullable(obj.getDefaultStyle())
+            .orElse(
+                new StyleNestedObject(Optional.ofNullable(obj.getColor()).orElse(FALLBACK_COLOR)));
     layer
         .setId(id)
-        .setListHeading(getLocalizedMessage(obj.getListHeading()))
-        .setItemLabel(getLocalizedMessage(obj.getItemLabel()))
-        .setDefaultStyle(
-            obj.getDefaultStyle() == null
-                ? DEFAULT_DEFAULT_STYLE
-                : StyleConverter.toStyle(obj.getDefaultStyle()));
+        .setName(getLocalizedMessage(obj.getName()))
+        .setDefaultStyle(StyleConverter.toStyle(style));
```

Comment:
`StyleNestedObject` is converted back to `Style` in the next step. Better to create instance of `Style` instead of `StyleNestefObject` in "orElse" block.

```
    Style style =
        Optional.ofNullable(obj.getDefaultStyle())
            .map(StyleConverter::toStyle)
            .orElse(
                Style.builder().setColor(Optional.ofNullable(obj.getColor()).orElse(FALLBACK_COLOR)).build());

    layer
        .setId(id)
        .setListHeading(getLocalizedMessage(obj.getListHeading()))
        .setItemLabel(getLocalizedMessage(obj.getItemLabel()))
        .setDefaultStyle(style);
```

Reply:
The two are functionally equivalent, and the cost of conversion is approx. 0, so I opted for creating a default input instead of output. Happy to change this, just wanted to mention it since in general we want to avoid over-optimizing code at the cost of readability.